src/Language/PureScript/CoreImp/ToJSON.hs

```diff
@@ -0,0 +1,163 @@
+{-# LANGUAGE NoOverloadedStrings #-}
+-- |
+-- Dump the core imperative representation in JSON format for consumption
+-- by third-party code generators
+--
+module Language.PureScript.CoreImp.ToJSON
+  ( Language.PureScript.CoreImp.ToJSON.moduleToJSON
+  ) where
+
+
+import Prelude.Compat
+
+import Data.Aeson
+import qualified Data.Text as T
+import Data.Version (Version)
+
+import Language.PureScript.AST (SourceSpan)
+import Language.PureScript.CoreFn (Module)
+import Language.PureScript.CoreFn.ToJSON as CoreFnToJSON
+import Language.PureScript.CoreImp.AST
+import Language.PureScript.PSString
+
+
+
+
+moduleToJSON :: (Show a) => Version -> Module a -> [AST] -> Value
+moduleToJSON v m a = object [ T.pack "body"    .= map (astToJSON) a
+                            , T.pack "corefn"  .= CoreFnToJSON.moduleToJSON v m -- might be good to "keep a snapshot of the original FP logic of the module" depending on coreimp-processing use-case (others can ignore it)
```

Comment:
I don't know how I feel about including the functional core here. What's the use case?

Reply:
Ultimately I want to push (later) for an argument "include-everything-you-know-incl-especially-type/ctor-infos" to `CoreFn.ToJSON.moduleToJSON`. I get that the existing `--dump-corefn` output is meant to be small and semi-readable for (typically) troubleshooting use-cases, but for the `--dump-coreimp` I see the use-cases as tools that need the most information possible about the original source. More in-depth type info (ADTs / records) in the current (JS-focused) CoreImp will be very hard to reconstruct from these "untyped" (other than stringlit/boollit/numlit/obj-lit) --- including a more detailed corefn (than the original/current `--dump-corefn` scenario envisions) right there in the coreimp dump will probably help. I guess for now that's too vague to let into the main repo so I'll take it out for now in the next commit, then keep tinkering locally until it all solidifies a bit more.